cg/meta/workflow/mutant/quality_controller/quality_controller.py

```diff
@@ -0,0 +1,128 @@
+from pathlib import Path
+from cg.apps.lims.api import LimsAPI
+from cg.constants.constants import MutantQC
+from cg.meta.workflow.mutant.quality_controller.models import (
+    QualityMetrics,
+    SampleQualityResult,
+    CaseQualityResult,
+    QualityResult,
+)
+
+from cg.meta.workflow.mutant.quality_controller.result_logger import ResultLogger
+from cg.meta.workflow.mutant.quality_controller.utils import (
+    has_valid_total_reads,
+    internal_negative_control_qc_pass,
+    external_negative_control_qc_pass,
+    get_quality_metrics,
+    get_report_path,
+)
+from cg.meta.workflow.mutant.quality_controller.report_generator import ReportGenerator
+from cg.store.models import Case
+from cg.store.store import Store
+
+
+class QualityController:
+    def __init__(self, status_db: Store, lims: LimsAPI) -> None:
+        self.status_db = status_db
+        self.lims = lims
+
+    def quality_control(
+        self, case: Case, case_path: Path, case_results_file_path: Path
+    ) -> QualityResult | None:
+        """Perform QC check on a case and generate the QC_report."""
+        quality_metrics: QualityMetrics = get_quality_metrics(
+            case_results_file_path, case, self.status_db, self.lims
+        )
+        if not quality_metrics:
+            return None
+        else:
+            sample_results: list[SampleQualityResult] = self.quality_control_samples(
+                quality_metrics
+            )
+            case_result: CaseQualityResult = self.quality_control_case(sample_results)
+
+            report_file: Path = get_report_path(case_path=case_path)
+            ReportGenerator.report(out_file=report_file, samples=sample_results, case=case_result)
+            ResultLogger.log_results(case=case_result, samples=sample_results, report=report_file)
```

Comment:
Change the occurences of the Constructor calls here also in other places. The reason for this is that the class needs the ReportGenerator to function, yet that dependency is not established in the constructor. 

Also, the code is as it is now is tied to this specific ReportGenerator, which makes it inflexible and also more difficult to test.

```suggestion
            self.report_generator.report(out_file=report_file, samples=sample_results, case=case_result)
            self.logger.log_results(case=case_result, samples=sample_results, report=report_file)
```

Reply:
That makes sense! Thanks!

Reply:
Instead of classes, have instead added the report_generator and report_logger as functions in utils files.